CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -154,37 +154,42 @@ function Staff:tick()
   self:updateSpeed()
 end
 
-function Staff:checkIfWaitedTooLong()
-  if self.quitting_in then
-    self.quitting_in = self.quitting_in - 1
-    if self.quitting_in < 0 then
-      local rise_windows = self.world.ui:getWindows(UIStaffRise)
-      local staff_rise_window = nil
+function Staff:checkIfWaitedTooLongForRaise()
+  if self.quitting_in >= 0 then return end
+  self.quitting_in = self.quitting_in - 1
 
-      -- We go through all "requesting rise" windows open, to see if we need
-      -- to close them when the person is fired.
-      for i = 1, #rise_windows do
-        if rise_windows[i].staff == self then
-          staff_rise_window = rise_windows[i]
-          break
-        end
-      end
-      -- If the hospital policy is set to automatically grant wage increases, grant
-      -- the requested raise instead of firing the staff member
-      if self.hospital.policies.grant_wage_increase then
-        local amount = math.floor(math.max(self.profile.wage * 1.1, (self.profile:getFairWage(self.world) + self.profile.wage) / 2) - self.profile.wage)
-        self.quitting_in = nil
-        self.hospital:removeMessage(self)
-        self:increaseWage(amount)
-        return
+  local is_waiting_time_is_up = self.quitting_in < 0
+  if is_waiting_time_is_up then
+    local rise_windows = self.world.ui:getWindows(UIStaffRise)
+    local staff_rise_window = nil
+    self.quitting_in = nil
+    self.hospital:removeMessage(self)
```

Comment:
This line specifically should still be moved down to the very very bottom of the function. There is currently no harm running this function call more than once but we shouldn't give instructions to destroy the window while we _may_ need it if it's open.